exp/slice/slice.go

```diff
@@ -21,6 +22,16 @@ func Take[A any](slice []A, n int) []A {
 	return slice[:n]
 }
 
+// Last returns the last element of a slice. If the slice is empty, it
+// returns the zero value.
+func Last[T any](list []T) T {
```

Comment:
I think I'll do something like:

```go
func Last[T any](list []T) (T, bool) {
	if len(list) == 0 {
		var zero T
		return zero, true
	}
	return list[len(list)-1], false
```

because its probably useful to know when the result is the zero value because that was the value in the slice vs when the slice is actually empty

Reply:
I considered that, and initially decided against because it makes it less convenient to use directly on scenarios like this:

```go
doSomething(slice.Last(list))
```

That now will have to be written as:

```go
item, ok := slice.Last(list)
if ok {
        doSomething(item)
}
```

But I'm okay with it, so I proceeded with the change.